src/awkward/_v2/contents/content.py

```diff
@@ -607,7 +607,7 @@ def _localindex_axis0(self):
     # FIXME: Temporary fix for concatenate (only working for axis = 0)
     def concatenate(self, others):
         if not isinstance(others, list):
-            others = [others]
+            others = [ak._v2.contents.NumpyArray([others])]
```

Comment:
With `concatenate` having been implemented, this method can be removed now, right? If it still has use as a helper function, then the "FIXME" can be removed, but also the arguments should be normalized to Content subclasses in the high-level function (which would ensure that any raw NumPy arrays are already converted to NumpyArray via a single call to `to_layouts`—by the time we get here, it should already be a Content subclass).